doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+finished contours and an array of face fully contours; in other words,
+outputs are, the first, an array of contours to be sharped and, the second,
+another one to be smoothed.
+
+To understand the points numeration we should just see the following
+illustration:
+
+![Landmarks order](pics/landmarks_illustration.png)
+
+The separate discussion of every element drawing follows.
+
+#### Eye contour getting {#gapi_fb_ld_eye}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp ld_pp_eye
+
+The function used to get the bottom side of an eye contour should be discussed:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_pp_eye
+
+Briefly, this function restores the bottom side of an eye by a half-ellipse
+based on two points in left and right eyecorners.
+
+As we can see, to minimize the frequency of memory allocations the
+`std::vector::reserve()` function is used; we can pass the real quantity of
+points to be stored in `cntEyeBottom` through the `capacity` argument if we know
+it apriory.
+
+Why isn't memory allocated at the beginning of the `getEyeEllipse()` function?
+The reason is the `cv::ellipse2Poly()` feature: instead of just filling out the
+given array the function assigns to it an array of points created inside;
+therefore, there is no sense to allocate memory before `cv::ellipse2Poly()`
+call.
+
+The rest of the `getEyeEllipse()` function is parameters preparation for the
+`cv::ellipse2Poly()` function call.
+
+![Ellipse2poly illustration](modules/imgproc/doc/pics/ellipse.svg)
+
+What is to be defined:
+- the ellipse center and the X half-axis counted by two eye Points;
+- the Y half-axis counted according to the assumption that an average eye width
+is 1/3 of its length;
+- the start and end angles which are 0 and 180 (see the picture above)
+- the angle delta: how frequently (which causes by how much points) the ellipse
+will be approximated;
+- the inclination angle of axes counted via the
+`getLineInclinationAngleDegrees()` function:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp toDbl
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_pp_incl
+
+The important thing is the use of the `atan2()` function, which can return
+negative value depending on the `x` and `y` signs; this fact allows to get the
+right angle even in case of upside-down face arrangment (if we put the points in
+the right order, of course).
+
+Back to the `getEyeEllipse()` function, it returns a constructed `Contour`
+to use the RVO.
+
+After the bottom side of an eye is approximated, we just push the eyebrow points
+to the contour in the right order - and the eye is finished!
+
+#### Mouth contour getting {#gapi_fb_ld_mth}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp ld_pp_mth
+
+We have four points of a mouth given: two of them in corners and two others in
```

Comment:
`the mouth`
`and the others two`

Reply:
OK

Reply:
@OrestChura Please stop spamming us ! I got **a lot of mails**

Reply:
> @OrestChura Please stop spamming us ! I got **a lot of mails**

@hannessa2 somehow you've got subscribed to this PR as a reviewer. Please unsubscribe (I couldn't find a button how to remove you from the reviewers list)